src/DssSpell.sol

```diff
@@ -48,36 +153,321 @@ contract DssSpellAction is DssAction {
+    uint256 internal constant WAD      = 10**18;
+    uint256 internal constant RAD      = 10**45;
 
-    // ---------- Contracts ----------
-    GemAbstract internal immutable MKR               = GemAbstract(DssExecLib.mkr());
+    // ---------- Phase 1b Addresses ----------
 
-    // ---------- Spark Proxy Spell ----------
-    // Spark Proxy: https://github.com/marsfoundation/sparklend-deployments/blob/bba4c57d54deb6a14490b897c12a949aa035a99b/script/output/1/primary-sce-latest.json#L2
-    address internal constant SPARK_PROXY = 0x3300f198988e4C9C63F75dF86De36421f06af8c4;
-    address internal constant SPARK_SPELL = 0x668C84584Ef8EeEd6BFb4FFB2a4Fa03231F8b241;
+    address internal constant USDS                         = 0xdC035D45d973E3EC169d2276DDab16f1e407384F;
+    address internal constant USDS_IMP                     = 0x1923DfeE706A8E78157416C29cBCCFDe7cdF4102;
+    address internal constant USDS_JOIN                    = 0x3C0f895007CA717Aa01c8693e59DF1e8C3777FEB;
+    address internal constant DAI_USDS                     = 0x3225737a9Bbb6473CB4a45b7244ACa2BeFdB276A;
+    address internal constant SUSDS                        = 0xa3931d71877C0E7a3148CB7Eb4463524FEc27fbD;
+    address internal constant SUSDS_IMP                    = 0x4e7991e5C547ce825BdEb665EE14a3274f9F61e0;
+    address internal constant SKY                          = 0x56072C95FAA701256059aa122697B133aDEd9279;
+    address internal constant MKR_SKY                      = 0xBDcFCA946b6CDd965f99a839e4435Bcdc1bc470B;
+    address internal constant UNIV2DAIMKR                  = 0x517F9dD285e75b599234F7221227339478d0FcC8;
+    address internal constant UNIV2USDSSKY                 = 0x2621CC0B3F3c079c1Db0E80794AA24976F0b9e3c;
+    address internal constant MCD_SPLIT                    = 0xBF7111F13386d23cb2Fba5A538107A73f6872bCF;
+    address internal constant SPLITTER_MOM                 = 0xF51a075d468dE7dE3599C1Dc47F5C42d02C9230e;
+    address internal constant MCD_FLAP                     = 0xc5A9CaeBA70D6974cBDFb28120C3611Dd9910355;
+    address internal constant FLAP_SKY_ORACLE              = 0x38e8c1D443f546Dc014D7756ec63116161CB7B25;
+    address internal constant MCD_VEST_SKY                 = 0xB313Eab3FdE99B2bB4bA9750C2DDFBe2729d1cE9;
+    address internal constant REWARDS_USDS_SKY             = 0x0650CAF159C5A49f711e8169D4336ECB9b950275;
+    address internal constant REWARDS_DIST_USDS_SKY        = 0x2F0C88e935Db5A60DDA73b0B4EAEef55883896d9;
+    address internal constant REWARDS_USDS_01              = 0x10ab606B067C9C461d8893c47C7512472E19e2Ce;
+    address internal constant CRON_REWARDS_DIST_JOB        = 0x6464C34A02DD155dd0c630CE233DD6e21C24F9A5;
+    address internal constant WRAPPER_USDS_LITE_PSM_USDC_A = 0xA188EEC8F81263234dA3622A406892F3D630f98c;
+
+    // ---------- MCD Addresses ----------
+    address internal immutable MCD_PAUSE = DssExecLib.getChangelogAddress("MCD_PAUSE");
 
     function actions() public override {
```

Comment:
Please add one missing instruction "Do not authorize ESM in any new contracts" + Note